File: 源码阅读/HZThreeParty/HZThreeParty/HZThreeParty/HZListViewController.m
//
//  HZListViewController.m
//  HZUIKIT
//
//  Created by apple on 16/9/1.
//  Copyright © 2016年 HZ. All rights reserved.
//

#import "HZListViewController.h"

@interface HZListViewController ()<UITableViewDelegate,UITableViewDataSource>
@end

@implementation HZListViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    [self.view setBackgroundColor:[UIColor whiteColor]];
    self.edgesForExtendedLayout = UIRectEdgeNone;
    [self setUpView];
}

-(void)viewWillAppear:(BOOL)animated{
    [super viewWillAppear:animated];
}

#pragma mark - private Method

-(void)setUpView{
    UITableView* tableView = [[UITableView alloc] initWithFrame:self.view.frame];
    tableView.delegate = self;
    tableView.dataSource = self;
    [self.view addSubview:tableView];
}

#pragma mark - UITableViewDataSource

-(NSInteger)numberOfSectionsInTableView:(UITableView *)tableView{
    return 1;
}

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return self.tableViewItems.count;
}

-(UITableViewCell*)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    static NSString* cellIdentifier = @"cellID";
    UITableViewCell* cell = [tableView dequeueReusableCellWithIdentifier:cellIdentifier];
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:cellIdentifier];
    }
    cell.textLabel.text = self.tableViewItems[indexPath.row];
    return cell;
}

#pragma mark - UITableViewDelegate

-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    UIViewController* viewController = [[NSClassFromString(self.tableViewItems[indexPath.row]) alloc] init];
    viewController.title = self.tableViewItems[indexPath.row];
    [self.navigationController pushViewController:viewController animated:YES];
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: 基础-语言篇/Swift/HZThreeParty/HZThreeParty/HZThreeParty/HZListViewController.swift
//
//  HZListViewController.swift
//  HZThreeParty
//
//  Created by KB on 2018/7/30.
//  Copyright © 2018年 BQ. All rights reserved.
//

import UIKit

class HZListViewController: UIViewController,UITableViewDelegate,UITableViewDataSource{
    public var dataSource: [String]?
    public var tableView: UITableView?

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        setUpSubView()
    }
    
    func setUpSubView() -> Void {
        tableView = UITableView(frame: self.view.frame, style: UITableViewStyle.plain)
        tableView!.delegate = self
        tableView!.dataSource = self
        self.view.addSubview(tableView!);
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return dataSource!.count;
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cellID = "threePartyIdentifier"
        var cell = tableView.dequeueReusableCell(withIdentifier: cellID)
        if cell == nil {
            cell = UITableViewCell(style: UITableViewCellStyle.default, reuseIdentifier: cellID)
        }
        cell?.textLabel?.text = dataSource![indexPath.row]
        return cell!
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let classString = (self.dataSource?[indexPath.row])!
        let  className:AnyClass? =   NSClassFromString("HZThreeParty." + classString)
        let viewControllerClass: UIViewController.Type = className as! UIViewController.Type
        let viewController = viewControllerClass.init()
        self.navigationController?.pushViewController(viewController, animated: true)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}
